### ISSUE TITLE
Save the model weight to null

### ISSUE THREAD

**Issue by zerodohero** [tensorpack/tensorpack#1481]
This is my environment
**Window server 2012
tensorflow-gpu==1.12.0
tensorpack==0.9.0.1
scikit-image==0.14.2
matplotlib===3.0.2
numpy==1.15.4
scipy==1.4.1
opencv-python==4.1.2.30
openslide-python==1.1.1**

This is the code that holds the weight of the model
 **print("self.save_dir",self.save_dir)
        if save_dir is None:
            logger.set_logger_dir(self.save_dir)
        else:
            logger.set_logger_dir(save_d**ir)

This is the result saved at the end of the model. The arrow points to the empty weight of the model, which is not regular. It will be more in the first few rounds, but it will be less as the number of training rounds increases
![image](https://user-images.githubusercontent.com/38395598/91919475-b327dc80-ecf8-11ea-9fba-ed5e583f2bcb.png)

This is because Windows does not support commented code,
def train_generator(ds, shape_aug=None, input_aug=None, label_aug=None, batch_size=16, nr_procs=8):
    ### augment both the input and label
    ds = ds if shape_aug is None else AugmentImageComponents(ds, shape_aug, (0, 1), copy=True)
    ### augment just the input i.e index 0 within each yield of DatasetSerial
    ds = ds if input_aug is None else AugmentImageComponent(ds, input_aug, index=0, copy=False)
    ### augment just the output i.e index 1 within each yield of DatasetSerial
    ds = ds if label_aug is None else AugmentImageComponent(ds, label_aug, index=1, copy=True)
    #
    **ds = BatchDataByShape(ds, batch_size, idx=0)
    #ds = PrefetchDataZMQ(ds, nr_procs)
    #ds = MultiProcessRunner(ds, 16,nr_procs)##_### Do you have any Suggestions about this replacement? I can switch to version 0.10.0 and use this line of code, but the program won't run, it will get stuck_
    return ds**

So this is the weight information that was saved eventually, the arrow points to no information was saved, so I was wondering is there any mechanism in tensorPack for that to happen, Or do you have any other Suggestions

**Comment by zerodohero** [comment]
This code is correct because the original author has no weight saved to 0 on Linux, the only difference between me and him is that I used Window, and I deleted #ds = PrefetchDataZMQ(DS, NR_procs), this line of code, because WinSOW doesn't support,

**Comment by ppwwyyxx** [comment]
We cannot help with it unless we can reproduce it on linux.
Changes to data loading should have no effect on saving. The cause is something else.
